lib/saluki-io/src/net/client/http.rs

```diff
@@ -62,7 +62,9 @@ where
 {
     pub fn from_connector(connector: C) -> Self {
         HttpClient {
-            inner: Client::builder(TokioExecutor::new()).build(connector),
+            inner: Client::builder(TokioExecutor::new())
+                .pool_max_idle_per_host(100)
```

Comment:
You're setting the overall maximum idle pool size value here, whereas in the Agent the _per-host_ value is [5](https://github.com/DataDog/datadog-agent/blob/0501086088f6615845ad1d052729cc6d27a23e56/pkg/util/http/transport.go#L117C3-L117C22).